Support kubernetes_role argument for prometheus.operator.servicemonitors (#2023)
grafana/alloy: docs/sources/reference/components/prometheus/prometheus.operator.servicemonitors.md

The following arguments are supported:

Name | Type | Description                                                                                                                | Default | Required
---- | ---- |----------------------------------------------------------------------------------------------------------------------------| ------- | --------
`forward_to` | `list(MetricsReceiver)` | List of receivers to send scraped metrics to.                                                                              | | yes
`namespaces` | `list(string)` | List of namespaces to search for ServiceMonitor resources. If not specified, all namespaces will be searched.              || no

> `kubernetes_role` | `string` | The Kubernetes role used for discovery. Supports `endpoints` or `endpointslice` If not specified, the `endpoints` role is used. || no

Reviewer: The "Default" column should have `endpoints` in it, right?

> `kubernetes_role` | `string` | The kubernetes role used for discovery. Supports `endpoints` or `endpointslice` If not specified, `endpoints` role will be used. || no

Reviewer 2: ```suggestion
`kubernetes_role` | `string` | The Kubernetes role used for discovery. Supports `endpoints` or `endpointslice` If not specified, the `endpoints` role is used. || no
```

> `kubernetes_role` | `string` | The Kubernetes role used for discovery. Supports `endpoints` or `endpointslice` If not specified, the `endpoints` role is used. | endpoints | no

Reviewer: ```suggestion
`kubernetes_role` | `string` | The Kubernetes role used for discovery. Supports `endpoints` or `endpointslice`. | `endpoints` | no
```

Reviewer 2: Ah good catch... the default is implied and we don't need to state it in the description.
